phpunit/directives/attributes/wp-show.php

```diff
@@ -0,0 +1,176 @@
+		$markup = '<p>Some text</p><div data-wp-show="context.myblock.open">I should not be shown!</div>';
+
+		$tags = new WP_Directive_Processor( $markup );
+		$tags->next_tag( 'div' );
+
+		$context_before = new WP_Directive_Context( array( 'myblock' => array( 'open' => false ) ) );
+		$context        = clone $context_before;
+		process_wp_show( $tags, $context );
+
+		$tags->next_tag( array( 'tag_closers' => 'visit' ) );
+		process_wp_show( $tags, $context );
+
+		$updated_markup = '<p>Some text</p><template data-wp-show="context.myblock.open"><div >I should not be shown!</div></template>';
+
+		$this->assertSame( $updated_markup, $tags->get_updated_html() );
+		$this->assertSame( $context_before->get_context(), $context->get_context(), 'data-wp-show directive changed context' );
+	}
+
+	public function test_directive_wraps_void_tag_in_template_if_when_is_false() {
+		$markup = '<img data-wp-show="context.myblock.open">';
+		$tags   = new WP_Directive_Processor( $markup );
+		$tags->next_tag();
+
+		$context_before = new WP_Directive_Context( array( 'myblock' => array( 'open' => false ) ) );
+		$context        = clone $context_before;
+		process_wp_show( $tags, $context );
+
+		$tags->next_tag( array( 'tag_closers' => 'visit' ) );
+		process_wp_show( $tags, $context );
+
+		$updated_markup = '<template data-wp-show="context.myblock.open"><img ></template>';
+
+		$this->assertSame( $updated_markup, $tags->get_updated_html() );
+		$this->assertSame( $context_before->get_context(), $context->get_context(), 'data-wp-show directive changed context' );
+	}
+
+	public function test_nested_directives_within_wp_show_with_truthy_value() {
+		$markup = <<<END
+			<div data-wp-show="context.myBlock.open">
+				I should be shown!
+				<div data-wp-show="context.myOtherBlock.open">I should be shown!</div>
+				<div data-wp-show="!context.myOtherBlock.open">I should not be shown!</div>
+			</div>
+END;
+
+		$context_before = new WP_Directive_Context(
+			array(
+				'myBlock'      => array( 'open' => true ),
+				'myOtherBlock' => array( 'open' => true ),
+			)
+		);
+		$context        = clone $context_before;
+
+		$tags = new WP_Directive_Processor( $markup );
+
+		while ( $tags->next_tag( array( 'tag_closers' => 'visit' ) ) ) {
+			process_wp_show( $tags, $context );
+		}
+
+		$updated_markup = <<<END
+			<div data-wp-show="context.myBlock.open">
+				I should be shown!
+				<div data-wp-show="context.myOtherBlock.open">I should be shown!</div>
+				<template data-wp-show="!context.myOtherBlock.open"><div >I should not be shown!</div></template>
+			</div>
+END;
+		$this->assertSame( $updated_markup, $tags->get_updated_html() );
+		$this->assertSame( $context_before->get_context(), $context->get_context(), 'data-wp-show directive changed context' );
+	}
+
+	public function test_nested_directives_within_wp_show_with_falsy_value() {
+		$markup = <<<END
+			<div data-wp-show="context.myBlock.open">
+				I should be shown!
```

Comment:
Nitpick: when using the falsy value, it should follow other tests and the text used:
```suggestion
				I should not be shown!
```

Reply:
The expected result needs to get updated, too. Otherwise the test fails.